ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKTable.java

```diff
@@ -224,12 +216,12 @@ public SchemaKGroupedTable groupBy(
       final Stacker contextStacker
   ) {
     final KeyFormat groupedKeyFormat = KeyFormat
-        .nonWindowed(keyFormat.getFormatInfo(), SerdeFeatures.of());
+        .nonWindowed(keyFormat.getFormatInfo(), keyFormat.getFeatures());
```

Comment:
This change is strange to me -- what was your motivation? This will have to be updated to support multi-column group by's, since the new schema (post group by) needs to be taken into account when deciding what key serde features to set. (I can do that as part of my group by PR, just curious why this change is being made here when it'll have to change again soon anyway.)

I also notice that the analogous change wasn't made in SchemaKStream.

Reply:
I think I answered my own question: there is no analogous change in SchemaKStream since stream group bys don't need key format. Defaulting to propagating key serde features in table group bys means fewer historic execution plans change, since internal topics today always unwrap single keys (which is what gets propagated with this change, else we'd get new plans with empty serde features after table group bys).

I don't think this inconsistency between stream and table group bys makes sense in the long run, but the logic here will change with the next PR anyway so I think it's fine.

Reply:
Yeah, I assumed that with multi-col group bys you'd need to generate a new key format and probably change the key features. But what's nice with this approach is that whatever mechanism you decide should be used for generating the key format will automagically be propagated down the line.